sciris/sc_fileio.py

```diff
@@ -945,7 +967,10 @@ def recursive_substitute(obj1, obj2, track=None):
                         k = k.decode('latin1')
                     track2 = track.copy()
                     track2.append(k)
-                    recursive_substitute(obj1[k], v, track2)
+                    if count<=maxcount:
+                        count = recursive_substitute(obj1[k], v, track2, count)
+                    else:
+                        print(recursion_warning(count, obj1, obj2))
```

Comment:
@RomeshA I thought it was nice to catch recursion before Python dies, but open to suggestions...

Reply:
@cliffckerr seems fine, I assume the idea with catching it is so that it can display a warning and make a failed class and load the rest of the file? I think the recursive calls probably need `maxcount` to be passed to them as well e.g. `count = recursive_substitute(obj1[k], v, track2, count,maxcount)`?

Reply:
@RomeshA Yes good point!